Automattic/wp-calypso
# Checkout: update support link

In #48695, we fixed the right margin of the Checkout chat link when inside the `CheckoutSummary`, but when the monthly plans rollout happened, the link was absolutely positioned below the summary, creating a number of display issues. This PR bundles the `CheckoutSupportLink` component (logic and styles) with the `PaymentsChatButton`, renames that files, and properly includes it one level up (inside of `CheckoutSummaryBody`), beneath the upsells (when visible).

Fixes: #49150

**Chat eligible:**
![Screen Shot 2021-01-27 at 11 51 31 AM](https://user-images.githubusercontent.com/942359/106027609-0e3be080-6099-11eb-8d84-7a49f21d14c6.png)

**Chat eligible (with upsell):**
![Screen Shot 2021-01-27 at 11 50 58 AM](https://user-images.githubusercontent.com/942359/106027378-d765ca80-6098-11eb-9a4c-e33be1a52060.png)

**Chat eligible on mobile (with upsell):**
<img width="324" alt="Screen Shot 2021-01-27 at 11 51 10 AM" src="https://user-images.githubusercontent.com/942359/106027472-ea789a80-6098-11eb-8ba7-cba14316fa0a.png">

**Chat not eligible:**
![Screen Shot 2021-01-27 at 11 56 35 AM](https://user-images.githubusercontent.com/942359/106027714-2b70af00-6099-11eb-9fc0-78bf74af4d94.png)

**To test:**
- force the pre-sales HappyChat button to render by setting [this](https://github.com/Automattic/wp-calypso/blob/fix/checkout-chat-button/client/components/happychat/button.jsx#L83) and [this](https://github.com/Automattic/wp-calypso/blob/fix/checkout-chat-button/client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx#L153) to true
- add a domain to your cart to visit Checkout
- verify that the button matches the screenshots above (with the upsell)
- click the upsell to add a plan to your cart
- verify that the button matches the screenshots above (without the upsell)
